### ISSUE TITLE
rez cant find packaged pip

### ISSUE THREAD

**Issue by alxkocic** [nerdvegas/rez#1194]
Hi,

We have python and pip as packages (separate). When using rez-pip, the packaged pip is not found.

The problem is [this line here](https://github.com/nerdvegas/rez/blob/d392c65bf63b4bca8106f938cec49144ba54e770/src/rez/pip.py#L220) which tries to get full version of pip but it can't because python run in subprocess does not get pip module. The -E flag disables PYTHONPATH which is how we add pip module though the package in the first place.

```
name = 'pip'

version = '21.3.1'

description = ''

requires = []

build_requires = []

private_build_requires = []

variants = [['platform-windows', 'python-3.7']]

def commands():
    env.PYTHONPATH.append("{this.root}/Lib/site-packages")
```

So basically rez finds both python and pip packages, resolves context, runs python executable in that context but takes out pip and than says "nope, didnt find it" 🤷‍♂️ 

Related issue: #706
Related PR: #757
